File: apps/web/src/types/document.ts
// Mirrors apps/api/app/schemas/document.py.

// 12-Jun bugs 231/235 — l1_call + l2_call (Pre-Sales) and trial_summary
// (Solutioning) added in migration 0078.
export type DocKind =
  | "mom" | "vpd" | "recording" | "transcript" | "email" | "other" | "contract"
  | "l1_call" | "l2_call" | "trial_summary";
export type AiStatus = "pending" | "processing" | "complete" | "failed";

export interface ExtractedEntities {
  people?: string[];
  decisions?: string[];
  action_items?: string[];
  dates?: string[];
  is_stub?: boolean;
}

export interface Document {
  id: string;
  account_id: string;
  kind: DocKind;
  // 04-Jun — contract-doc subclassification (Signed Proposal / MSA / etc.)
  contract_subtype: string | null;
  filename: string;
  mime_type: string | null;
  size_bytes: number | null;
  meeting_date: string | null;
  uploaded_by: string | null;
  uploaded_at: string;
  ai_status: AiStatus;
  ai_summary_text: string | null;
  extracted_entities: ExtractedEntities | null;
  job_id: string | null;
  ai_edited: boolean;
  ai_edited_at: string | null;
  mom_extracted_fields: Record<string, unknown> | null;
  mom_extracted_at: string | null;
  vpd_extracted_fields: Record<string, unknown> | null;
  vpd_extracted_at: string | null;
  handoff_extracted_fields: Record<string, unknown> | null;
  handoff_extracted_at: string | null;
  cs_goals_extracted: Record<string, unknown> | null;
  cs_goals_extracted_at: string | null;
  metrics_extracted: Record<string, unknown> | null;
  metrics_extracted_at: string | null;
  notes: string | null;
  deleted_at: string | null;
}

export interface DocumentListResponse {
  items: Document[];
  total: number;
  is_editable: boolean;
}

export interface DocumentUploadResponse {
  document: Document;
  job_id: string;
  duplicate: boolean;
}

export interface Job {
  id: string;
  kind: string;
  account_id: string | null;
  document_id: string | null;
  status: "pending" | "running" | "complete" | "failed";
  progress: number;
  error: string | null;
  result: Record<string, unknown> | null;
  started_at: string | null;
  finished_at: string | null;
  created_at: string;
}

export interface DiscoverySummary {
  account_id: string;
  summary_text: string | null;
  source_document_ids: string[];
  generated_at: string | null;
}

export const DOC_KIND_LABELS: Record<DocKind, string> = {
  mom: "Meeting Minutes",
  vpd: "Value Proposition Deck",
  recording: "Recording",
  transcript: "Transcript",
  email: "Email",
  other: "Other",
  contract: "Contract",
  l1_call: "L1 Call Notes",
  l2_call: "L2 Call Notes",
  trial_summary: "Trial Summary",
};

export const AI_STATUS_LABELS: Record<AiStatus, string> = {
  pending: "Queued",
  processing: "Processing",
  complete: "Ready",
  failed: "Failed",
};

File: apps/api/app/schemas/document.py
"""AK03.c — Pydantic schemas for documents, jobs, and aggregate summary."""

from __future__ import annotations

from datetime import date, datetime
from typing import Any, Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

DocKind = Literal["mom", "vpd", "recording", "transcript", "email", "other", "contract"]
AiStatus = Literal["pending", "processing", "complete", "failed"]


class DocumentOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    account_id: UUID
    kind: DocKind
    # 04-Jun — contract-doc subclassification (Signed Proposal / MSA / etc.)
    contract_subtype: str | None = None
    filename: str
    mime_type: str | None
    size_bytes: int | None
    meeting_date: date | None
    uploaded_by: UUID | None
    uploaded_at: datetime
    ai_status: AiStatus
    ai_summary_text: str | None
    extracted_entities: dict[str, Any] | None
    job_id: UUID | None
    ai_edited: bool = False
    ai_edited_at: datetime | None = None
    mom_extracted_fields: dict[str, Any] | None = None
    mom_extracted_at: datetime | None = None
    vpd_extracted_fields: dict[str, Any] | None = None
    vpd_extracted_at: datetime | None = None
    handoff_extracted_fields: dict[str, Any] | None = None
    handoff_extracted_at: datetime | None = None
    cs_goals_extracted: dict[str, Any] | None = None
    cs_goals_extracted_at: datetime | None = None
    metrics_extracted: dict[str, Any] | None = None
    metrics_extracted_at: datetime | None = None
    notes: str | None = None
    deleted_at: datetime | None


class DocumentSummaryUpdate(BaseModel):
    """User edit of an AI summary — flips ai_edited to true server-side."""

    ai_summary_text: str = Field(..., min_length=1, max_length=4000)


class DocumentNotesUpdate(BaseModel):
    """Bug 3 — free-text notes attached to the uploaded document.
    Empty string allowed (clears the note)."""

    notes: str = Field("", max_length=4000)


# 04-Jun — Sales Handoff prototype port.
ContractSubtype = Literal[
    "Signed Proposal", "Unsigned Proposal", "Work Order",
    "MSA", "Purchase Order", "Invoice", "SOW",
    "Amendment", "NDA",
]


class DocumentContractSubtypeUpdate(BaseModel):
    """PATCH body for contract-doc subtype tagging.

    Sales picks one of the 9 canonical types from a dropdown after
    upload; the audit table then filters/sorts by type. Server-side
    CHECK constraint bounds the set."""

    contract_subtype: ContractSubtype | None = None


class DocumentListResponse(BaseModel):
    items: list[DocumentOut]
    total: int
    is_editable: bool


class DocumentUploadResponse(BaseModel):
    """Returned by POST /api/v1/accounts/:id/documents — 202 Accepted."""

    document: DocumentOut
    job_id: UUID
    duplicate: bool = Field(
        False,
        description="True when the file hash matched an existing doc; we returned that one instead.",
    )


class JobOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    kind: str
    account_id: UUID | None
    document_id: UUID | None
    status: Literal["pending", "running", "complete", "failed"]
    progress: int
    error: str | None
    result: dict[str, Any] | None
    started_at: datetime | None
    finished_at: datetime | None
    created_at: datetime


class DiscoverySummaryOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    account_id: UUID
    summary_text: str | None
    source_document_ids: list[UUID]
    generated_at: datetime | None
